src/main/java/com/likelion/backendplus4/talkpick/batch/news/article/infrastructure/collector/support/scraper/implement/KhanContentScraper.java

```diff
@@ -27,168 +29,197 @@ public class KhanContentScraper implements ContentScraper {
      *
      * @param url 기사 URL
      * @return 문단 단위로 나눈 본문 리스트
+     * @throws ArticleCollectorException 스크래핑 중 오류 발생 시
      */
     @Override
-    public List<String> scrapeParagraphs(String url) {
-        try {
-            Document document = connectToUrl(url);
-            List<String> paragraphs = extractKhanContent(document);
-            return paragraphs;
-        } catch (Exception e) {
-            throw new ArticleCollectorException(ArticleCollectorErrorCode.ITEM_MAPPING_ERROR, e);
+    public List<String> scrapeParagraphs(String url) throws ArticleCollectorException {
+        Document document = connectToUrl(url);
+        List<String> content = extractKhanContent(document);
+
+        if (content == null || content.isEmpty() || content.stream().allMatch(String::isEmpty)) {
+            throw new ArticleCollectorException(ArticleCollectorErrorCode.EMPTY_ARTICLE_CONTENT);
         }
+
+        return content;
     }
 
     /**
      * 경향신문 본문 추출 (article.art_body에서 h3, div.art_photo 제외)
      *
      * @param document JSoup Document
      * @return 문단 리스트
+     * @throws ArticleCollectorException 본문 파싱 중 오류 발생 시
      */
-    private List<String> extractKhanContent(Document document) {
-        Element artBody = HtmlScraperUtils.findElement(document, "article.art_body");
+    private List<String> extractKhanContent(Document document) throws ArticleCollectorException {
+        try {
+            Element artBody = HtmlScraperUtils.findElement(document, "article.art_body");
 
-        if (artBody == null) {
-            artBody = HtmlScraperUtils.findElement(document, "div.art_body");
-        }
+            if (artBody == null) {
+                artBody = HtmlScraperUtils.findElement(document, "div.art_body");
+            }
 
-        if (artBody == null) {
-            artBody = HtmlScraperUtils.findElement(document, "div.article_view");
-        }
+            if (artBody == null) {
+                artBody = HtmlScraperUtils.findElement(document, "div.article_view");
+            }
 
-        if (artBody == null) {
-            artBody = HtmlScraperUtils.findElement(document, "div.article-body");
-        }
+            if (artBody == null) {
+                artBody = HtmlScraperUtils.findElement(document, "div.article-body");
+            }
 
-        if (artBody == null) {
-            return new ArrayList<>();
-        }
+            if (artBody == null) {
+                return new ArrayList<>();
+            }
 
-        return extractKhanContentFromElement(artBody);
+            return extractKhanContentFromElement(artBody);
+        } catch (Exception e) {
+            throw new ArticleCollectorException(ArticleCollectorErrorCode.SCRAPER_PARSING_ERROR, e);
+        }
     }
```

Comment:
P2:
아래와 같이 리팩토링해보시는 것은 어떠신가요? List는 위로 올려 별도로 관리하면 좋을 것 같습니다.
```suggestion
private List<String> extractKhanContent(Document document) throws ArticleCollectorException {
    try {
        List<String> selectors = List.of(
            "article.art_body",
            "div.art_body",
            "div.article_view",
            "div.article-body"
        );

        // 가장 먼저 매칭되는 Element 하나만 찾기
        Element artBody = selectors.stream()
            .map(sel -> HtmlScraperUtils.findElement(document, sel))
            .filter(Objects::nonNull)
            .findFirst()
            .orElse(null);

        // 못 찾았으면 빈 리스트 반환
        if (artBody == null) {
            return new ArrayList<>();
        }

        // 찾았으면 실제 파싱 로직 호출
        return extractKhanContentFromElement(artBody);
    } catch (Exception e) {
        throw new ArticleCollectorException(
            ArticleCollectorErrorCode.SCRAPER_PARSING_ERROR, e
        );
    }
}

```

Reply:
findFirst를 이럴 때 써야되겠네요 👍

Reply:
좋은거 같습니다 Khan외에 앞서 지적해주신 Donga스크래퍼에도 같은 방식으로 리팩토링을 적용하겠습니다.